WooCommerce/Classes/ViewRelated/Products/Variations/ProductVariationsViewController.swift

```diff
@@ -536,6 +540,7 @@ private extension ProductVariationsViewController {
             let variationsUpdated = self.product.variations.filter { $0 != variation.productVariationID }
             let updatedProduct = self.product.copy(variations: variationsUpdated)
             self.product = updatedProduct
+            self.configureNavigationBarButtons()
```

Comment:
I think you can move this and the update below(when creating the variation) to a place that always is called when the dataset changes. Like the `configureResultsControllerEventHandling` method which has a `onReload` closure.

Reply:
That was my initial attempt. But when the `onReload` is called the `self.product` it is not yet updated and contains the old variation list. The `self.product` is updated when the `onVariationDeletion` or `onVariationCreation` are called. I also though to put the `self.configureNavigationBarButtons()` in the a property observer of the `product` but this will result in unnecessary calls to `self.configureNavigationBarButtons()`

Reply:
Hmm we don't need to check `self.product.variations.isNotEmpty`(note to unnecessary self) tho, we could check `resultsController.fetchedObjects.isNotEmpty` which is our real source of truth.

Reply:
True, updated the PR!